### PR TITLE
Add support for managing authentication tokens to RESTful API

### PR DESCRIPTION
**Pull request by Diman0**
## What type of PR?

enhancement

## What does this PR do?
Authentication tokens could not be managed yet via the RESTful API. Support for this has been added to the RESTful API. It also contains a small bug fix for updating domains. 

### Related issue(s)
n/a1

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

File: core/admin/mailu/api/v1/token.py [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        if 'comment' in data:
+            token_new.comment = data['comment']
+        if 'AuthorizedIP' in data:
+            token_new.ip = data['AuthorizedIP'].replace(' ','').split(',')
+        raw_password = pwd.genword(entropy=128, length=32, charset="hex")
+        token_new.set_password(raw_password)
+        models.db.session.add(token_new)
+        #apply the changes
+        db.session.commit()
+        flask.current_app.logger.info(f'token_new.id == {token_new.id}.')
```
Comment (nextgens) [inline core/admin/mailu/api/v1/token.py:81]: Hmm, this is not useful IMHO. What about a debug log statement that shows the request/response instead?

File: core/admin/mailu/api/v1/token.py [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        if 'comment' in data:
+            token_new.comment = data['comment']
+        if 'AuthorizedIP' in data:
+            token_new.ip = token_new.ip = data['AuthorizedIP'].replace(' ','').split(',')
+        raw_password = pwd.genword(entropy=128, length=32, charset="hex")
+        token_new.set_password(raw_password)
+        models.db.session.add(token_new)
+        #apply the changes
+        db.session.commit()
+        flask.current_app.logger.info(f'token_new.id == {token_new.id}.')
```
Comment (nextgens) [inline core/admin/mailu/api/v1/token.py:135]: ditto

File: core/admin/mailu/api/v1/token.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    'comment': fields.String(description='A description for the token. This description is shown on the Authentication tokens page', example='my comment'),
+    'AuthorizedIP': fields.String(description='Comma separated list of white listed IP addresses or networks that may use this token', example='88.77.66.55', attribute='ip')
+})
+
+token_user_post_response = api.model('TokenPostResponse', {
+    'id': fields.String(description='The record id of the token (unique identifier)', example='1'),
+    'token': fields.String(description='The created authentication token for the user.', example='2caf6607de5129e4748a2c061aee56f2', attribute='password'),
+    'email': fields.String(description='The email address of the user', example='John.Doe@example.com', attribute='user_email'),
+    'comment': fields.String(description='A description for the token. This description is shown on the Authentication tokens page', example='my comment'),
+    'AuthorizedIP': fields.String(description='Comma separated list of white listed IP addresses or networks that may use this token', example='88.77.66.55', attribute='ip'),
```
Comment (nextgens) [inline core/admin/mailu/api/v1/token.py:39]: https://www.rfc-editor.org/rfc/rfc5737.html#section-3

Comment (nextgens) [reply core/admin/mailu/api/v1/token.py:39]: same on the 3 others above btw ^

**Review by nextgens — CHANGES_REQUESTED** [review]

File: core/admin/mailu/api/v1/token.py (hunk 1 above)
Comment (Diman0) [reply core/admin/mailu/api/v1/token.py:81]: I forgot to remove this. I used this for debugging.

File: core/admin/mailu/api/v1/token.py (hunk 2 above)
Comment (Diman0) [reply core/admin/mailu/api/v1/token.py:135]: I forgot to remove this. I used this for debugging.

File: core/admin/mailu/api/v1/token.py (hunk 3 above)
Comment (Diman0) [reply core/admin/mailu/api/v1/token.py:39]: When I read that RFC, I derive I should only use one for those three blocks. Is it your hobby to read & memorize all RFC's?

Comment (nextgens) [reply core/admin/mailu/api/v1/token.py:39]: Hobby definitely not, ... but it is a requirement of the day job to understand how things work... and RFCs are usually a good start :)

Let's use IPs from the allocated test-ranges where possible

**Comment by Diman0** [comment]
bors try

**Review by nextgens — APPROVED** [review]
LGTM

**Pull request merged**